### ISSUE TITLE
Internal USB Not Part of USB Boot Order Group

### ISSUE THREAD

**Issue by circlenaut** [pcengines/seabios#51]
I have a USB DOM connected to the internal USB header on my APU4d4, a 4TB seagate HDD connected to the internal SATA port and another 4TB seagate HDD connected to the mSATA pcie port through a SATA to mSATA converter module.

I'd like always boot first from the USB DOM, unfortunately, even with USB being 1st in boot order, the DOM always boots appears after the HDDs according to the F10 boot menu. If I plug a regular USB drive through the back USB ports, it appears 1st as it should on the boot menu. 

Instead of placing the internal USB header as part of the USB boot group, would it be possible to create a separate boot group for it? i.e. "Internal USB"

**Comment by miczyg1** [comment]
Previously we had a division into internal and external. But the final decision was to unify the USB entries.

Bootorder has entries for all possible ports, however when a hub is attached, the device path which represents the connected device changes. it point to the USB device via a HUB for example. The bootorder is defined only for direct USB storage devices plugged into the ports.

I suspect that your DOM probabyl has some USB bridge/hub midway to the disk which causes the bootorder mismatch for this device. Given that, the bootorder list for USB devices can be almost infinitely extended, due to the USB architecture. 

I can provide a debug binary which should print the device path for your device. based on that I can prepare a fix.

**Comment by circlenaut** [comment]
I suspect you're right:

`root@host:~ # usbconfig
ugen1.1: <AMD EHCI root HUB> at usbus1, cfg=0 md=HOST spd=HIGH (480Mbps) pwr=SAVE (0mA)
ugen0.1: <0x1022 XHCI root HUB> at usbus0, cfg=0 md=HOST spd=SUPER (5.0Gbps) pwr=SAVE (0mA)
ugen2.1: <AMD EHCI root HUB> at usbus2, cfg=0 md=HOST spd=HIGH (480Mbps) pwr=SAVE (0mA)
ugen1.2: <vendor 0x0438 product 0x7900> at usbus1, cfg=0 md=HOST spd=HIGH (480Mbps) pwr=SAVE (100mA)
ugen2.2: <vendor 0x0438 product 0x7900> at usbus2, cfg=0 md=HOST spd=HIGH (480Mbps) pwr=SAVE (100mA)
ugen2.3: <Atrue UDOM> at usbus2, cfg=0 md=HOST spd=HIGH (480Mbps) pwr=ON (500mA)
ugen0.2: <Norelsys NS1066> at usbus0, cfg=0 md=HOST spd=SUPER (5.0Gbps) pwr=ON (2mA)
`

`root@host:~ # camcontrol devlist
<ST4000LM024-2AN17V 0001>          at scbus0 target 0 lun 0 (pass0,ada0)
<ST4000LM024-2AN17V 0001>          at scbus1 target 0 lun 0 (pass1,ada1)
<Atrue UDOM 1100>                  at scbus2 target 0 lun 0 (da0,pass2)
<ATA WDC WD101KRYZ-01 1H01>        at scbus3 target 0 lun 0 (da1,pass3)`

**Comment by miczyg1** [comment]
@circlenaut  just to be sure, could you please flash this debug build?
[apu4 v4.11.0.3 debug](https://cloud.3mdeb.com/index.php/s/wL5eRm9eczGYkd5/download)

It will print more verbose information from SeaBIOS on the COM1 and should indicate the device path of the DOM. Please attach the boot log if possible.

**Comment by circlenaut** [comment]
```
picocom v3.1

port is        : /dev/ttyUSB0
flowcontrol    : none
baudrate is    : 115200
parity is      : none
databits are   : 8
stopbits are   : 1
escape is      : C-a
local echo is  : no
noinit is      : no
noreset is     : yes
hangup is      : no
nolock is      : yes
send_cmd is    : sz -vv
receive_cmd is : rz -vv -E
imap is        : 
omap is        : 
emap is        : crcrlf,delbs,
logfile is     : none
initstring     : none
exit_after is  : not set
exit is        : no

Type [C-a] [C-h] to see available commands
Terminal ready
PC Engines apu4
coreboot build 20202901
BIOS version v4.11.0.3
4080 MB ECC DRAM

SeaBIOS (version rel-1.12.1.3-0-g300e8b7)
BUILD: gcc: (coreboot toolchain v1.52 June 11th, 2018) 8.1.0 binutils: (GNU Binutils) 2.30
SeaBIOS (version rel-1.12.1.3-0-g300e8b7)
BUILD: gcc: (coreboot toolchain v1.52 June 11th, 2018) 8.1.0 binutils: (GNU Binutils) 2.30
Found coreboot cbmem console @ cffde000
Found mainboard PC Engines apu4
Relocating init from 0x000e7b00 to 0xcfe55640 (size 39200)
Found CBFS header at 0xff800238
multiboot: eax=cfef0700, ebx=cfef06c4
boot order:
1: /pci@i0cf8/usb@10/usb-*@1
2: /pci@i0cf8/usb@10/usb-*@2
3: /pci@i0cf8/usb@10/usb-*@3
4: /pci@i0cf8/usb@10/usb-*@4
5: /pci@i0cf8/usb@12/usb-*@1
6: /pci@i0cf8/usb@12/usb-*@2
7: /pci@i0cf8/usb@12/usb-*@3
8: /pci@i0cf8/usb@12/usb-*@4
9: /pci@i0cf8/usb@13/usb-*@1
10: /pci@i0cf8/usb@13/usb-*@2
11: /pci@i0cf8/usb@13/usb-*@3
12: /pci@i0cf8/usb@13/usb-*@4
13: /pci@i0cf8/*@14,7
14: /pci@i0cf8/*@11/drive@0/disk@0
15: /pci@i0cf8/*@11/drive@1/disk@0
16: /pci@i0cf8/pci-bridge@2,5/*@0/drive@0/disk@0
17: /pci@i0cf8/pci-bridge@2,5/*@0/drive@1/disk@0
18: /rom@genroms/pxe.rom
19: pxen0
20: scon1
21: usben1
22: uartc1
23: uartd1
24: ehcien0
25: mpcie2_clk0
26: com2en0
27: boosten1
28: sd3mode0
29: iommu0
30: watchdog0000
31: 
Found 26 PCI devices (max PCI bus is 05)
Copying SMBIOS entry point from 0xcfe9f000 to 0x000f1400
Copying ACPI RSDP from 0xcfea0000 to 0x000f13d0
Copying MPTABLE from 0xcfec4000/cfec4010 to 0x000f11b0
Copying PIR from 0xcfec5000 to 0x000f1180
Using pmtimer, ioport 0x818
Scan for VGA option rom
No VGA found, scan for other display
sercon: using ioport 0x3f8
sercon: configuring as primary display
Turning on vga text mode console
c[?7l[2J[0mSeaBIOS (version rel-1.12.1.3-0-g300e8b7XHCI init on dev 00:10.0: regs @ 0xf7f22000, 4 ports, 32 slots, 32 byte contexts
XHCI    extcap 0x1 @ 0xf7f22500
XHCI    protocol USB  3.00, 2 ports (offset 1), def 0
XHCI    protocol USB  2.00, 2 ports (offset 3), def 10
XHCI    extcap 0xa @ 0xf7f22540
EHCI init on dev 00:13.0 (regs=0xf7f26020)
AHCI controller at 00:11.0, iobase 0xf7f25000, irq 0
Searching bootorder for: /pci@i0cf8/*@14,7
Found 4 serial ports
Searching bootorder for: /rom@img/setup
Searching bootorder for: /rom@img/memtest
)
Searching bootorder for: /pci@i0cf8/*@11/drive@0/disk@0
AHCI/0: Set transfer mode to UDMA-6
AHCI/0: registering: "AHCI/0: ST4000LM024-2AN17V ATA-10 Hard-Disk (3726 GiBytes)"
Searching bootorder for: /pci@i0cf8/*@11/drive@1/disk@0
AHCI/1: Set transfer mode to UDMA-6
AHCI/1: registering: "AHCI/1: ST4000LM024-2AN17V ATA-10 Hard-Disk (3726 GiBytes)"
Searching bootorder for: /pci@i0cf8/usb@13/hub@1/storage@2/*@0/*@0,0
Searching bootorder for: /pci@i0cf8/usb@13/hub@1/usb-*@2
XHCI port #4: 0x00200a03, powered, enabled, pls 0, speed 2 [Low]
USB MSC vendor='Atrue' product='UDOM' rev='1100' type=0 removable=1
USB MSC blksize=512 sectors=31703040
USB keyboard initialized
Initialized USB HUB (1 ports used)
scard_waitw: 0xf7f27026 1 1f2
WARNING - Timeout at sdcard_waitw:162!
All threads complete.
Scan for option roms

Press F10 key now for boot menu

Select boot device:

1. AHCI/0: ST4000LM024-2AN17V ATA-10 Hard-Disk (3726 GiBytes)
2. AHCI/1: ST4000LM024-2AN17V ATA-10 Hard-Disk (3726 GiBytes)
3. USB MSC Drive Atrue UDOM 1100
4. Payload [setup]
5. Payload [memtest]
Searching bootorder for: HALT
drive 0x000f1090: PCHS=0/0/0 translation=lba LCHS=1024/255/63 s=31703040
drive 0x000f1110: PCHS=16383/16/63 translation=lba LCHS=1024/255/63 s=3519069872
drive 0x000f10c0: PCHS=16383/16/63 translation=lba LCHS=1024/255/63 s=3519069872
Space available for UMB: c0000-ee800, f0000-f1090
Returned 249856 bytes of ZoneHigh
e820 map has 7 items:
  0: 0000000000000000 - 000000000009fc00 = 1 RAM
  1: 000000000009fc00 - 00000000000a0000 = 2 RESERVED
  2: 00000000000f0000 - 0000000000100000 = 2 RESERVED
  3: 0000000000100000 - 00000000cfe9c000 = 1 RAM
  4: 00000000cfe9c000 - 00000000d0000000 = 2 RESERVED
  5: 00000000f8000000 - 00000000fc000000 = 2 RESERVED
  6: 0000000100000000 - 000000012f000000 = 1 RAM

enter handle_19:
  NULL
Booting from Hard Disk..Booting from 0000:7c00
.
-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-

Consoles: internal videoo[16;25H/keyboard   
 
BIOS drive C: is disk0 
 
BIOS drive D: is disk1 
 
BIOS drive E: is disk2 
 
BIOS 639kB/3405424kB available memory 
 
 
FreeBSD/x86 bootstrap loader, Revision 1.1 
 
| 
/ 
 
- 
\ 
| 
/ 
- 
\ 
| 
/ 
- 
\ 
| 
/ 
- 
\ 
 
| 
/ 
- 
\ 
 
| 
/ 
- 
\ 
 
| 
/ 
- 
 
\ 
| 
/ 
- 
 
\ 
| 
 
/ 
- 
\ 
 
| 
/ 
- 
 
\ 
| 
/ 
- 
 
\ 
| 
/ 
- 
 
\ 
| 
 
/ 
- 
\ 
 
| 
/ 
- 
\ 
| 
/ 
- 
\ 
| 
/ 
- 
\ 
| 
/ 
- 
\ 
 
| 
/ 
- 
\ 
 
Loading /boot/defaults/loader.conf 
 
| 
 
/ 
- 
\ 
| 
/ 
 
- 
\ 
| 
 
/ 
- 
\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-[H[J\|/-\| [9;48H[37;1m  @@@@@@@@@@@@@@@@@@@@@@@@@@@@[10;48H@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@[11;48H@@@@@                    @@@@@[12;48H    @@@@@            @@@@@    [13;48H @@@@@@@@@@@       @@@@@@@@@@@[14;48H      [31;1m\\\\\         /////     [15;48H))))))))))))       ((((((((((([16;48H      /////         \\\\\     [m[17;48H[37;1m @@@@@@@@@@@       @@@@@@@@@@@[18;48H    @@@@@            @@@@@    [19;48H@@@@@                    @@@@@[20;48H@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@[21;48H@@@@@@@@@@@@@@@@@@@@@@@@@@@@  [22;48H [m                              [23;48H   20.1 ``Keen Kingfisher''   [m/-\| [2;18H ______  _____  _____                         [3;18H/  __  |/ ___ |/ __  |                        [4;18H| |  | | |__/ | |  | |___  ___ _ __  ___  ___ [5;18H| |  | |  ___/| |  | / __|/ _ \ '_ \/ __|/ _ \[6;18H| |__| | |    | |  | \__ \  __/ | | \__ \  __/[7;18H|_____/|_|    |_| /__|___/\___|_| |_|___/\___|[10;2H|[11;2H|[12;2H|[13;2H|[14;2H|[15;2H|[16;2H|[17;2H|[18;2H|[19;2H|[20;2H|[21;2H|[10;44H|[11;44H|[12;44H|[13;44H|[14;44H|[15;44H|[16;44H|[17;44H|[18;44H|[19;44H|[20;44H|[21;44H|[9;3H=========================================[22;3H=========================================[9;2H+[22;2H+[9;44H+[22;44H+[25;0H/-\|/-\|/-\|/-\|/[9;24H[11;5H1 [11;6H.[11;8HBoot Multi User [Enter][12;5H2 [12;6H.[12;8HBoot [S]ingle User[13;5H3 [13;6H.[13;8H[Esc]ape to loader prompt[14;5H4 [14;6H.[14;8HReboot[16;5HOptions:[17;5H5 [17;6H.[17;8H[K]ernel: kernel (1 of 2)[18;5H6 [18;6H.[18;8HConfigure Boot [O]ptions...[25;0H[23;4HAutoboot in 2 seconds. [Space] to pause[25;0H[23;4HAutoboot in 1 seconds. [Space] to pause[25;0H[23;4H                                       [25;0H-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\/boot/kernel/kernel text=0x16dcd28 |/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\data=0x1520b8+0x534e18 |/-\|/-\|/-syms=[0x8+0x183be8\|/-\|/-\|/-+0x8+0x19cd09\|/-\|/-\|/-\]
|/-\|/-\|//boot/entropy -size=0x1000
\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/-\|/boot/kernel/carp.ko size 0xe248 at 0x2486000
/-\|/-\|/-\|/-\|/-\/boot/kernel/if_bridge.ko size 0xe4e8 at 0x2495000
loading required module 'bridgestp'
|/-\|/-\|/-\|/-\|/-/boot/kernel/bridgestp.ko size 0x6cb0 at 0x24a4000
\|/-\|/-\|/-\|//boot/kernel/if_enc.ko size 0x3250 at 0x24ab000
-\|/-\|/-\|/-\|/-\|/-/boot/kernel/if_gre.ko size 0x6e30 at 0x24af000
\|/-\|/-\|/-\|//boot/kernel/if_lagg.ko -size 0x14788 at 0x24b6000
\|/-\|/-\|/-\|/-\|//boot/kernel/if_tap.ko size 0x7bb8 at 0x24cb000
-\|/-\|/-\|/-\|/boot/kernel/pf.ko /-\size 0x55050 at 0x24d3000
|/-\|/-\|/-\|/-\|/-\|/boot/kernel/pflog.ko size 0x2ab8 at 0x2529000
/-\|/-\|/-\|/-\/boot/kernel/pfsync.ko |size 0xd590 at 0x252c000
/-\|/-\|/-\|/-\/boot/kernel/zfs.ko |/-\|/-\|/-\|/-\|/-\|/-\size 0x3810f0 at 0x253a000
loading required module 'opensolaris'
|/-\|/-\|/-\|/-/boot/kernel/opensolaris.ko size 0xa308 at 0x28bc000
Booting...
\|/-\|KDB: debugger backends: ddb
KDB: current backend: ddb
Copyright (c) 2013-2018 The HardenedBSD Project.
Copyright (c) 1992-2018 The FreeBSD Project.
Copyright (c) 1979, 1980, 1983, 1986, 1988, 1989, 1991, 1992, 1993, 1994
	The Regents of the University of California. All rights reserved.
FreeBSD is a registered trademark of The FreeBSD Foundation.
FreeBSD 11.2-RELEASE-p16-HBSD  fc65add89c3(stable/20.1) amd64
FreeBSD clang version 6.0.0 (tags/RELEASE_600/final 326565) (based on LLVM 6.0.0)
VT(vga): resolution 640x480
HardenedBSD: initialize and check features (__HardenedBSD_version 1100056 __FreeBSD_version 1102000).
CPU: AMD GX-412TC SOC                                (998.15-MHz K8-class CPU)
  Origin="AuthenticAMD"  Id=0x730f01  Family=0x16  Model=0x30  Stepping=1
  Features=0x178bfbff<FPU,VME,DE,PSE,TSC,MSR,PAE,MCE,CX8,APIC,SEP,MTRR,PGE,MCA,CMOV,PAT,PSE36,CLFLUSH,MMX,FXSR,SSE,SSE2,HTT>
  Features2=0x3ed8220b<SSE3,PCLMULQDQ,MON,SSSE3,CX16,SSE4.1,SSE4.2,MOVBE,POPCNT,AESNI,XSAVE,OSXSAVE,AVX,F16C>
  AMD Features=0x2e500800<SYSCALL,NX,MMX+,FFXSR,Page1GB,RDTSCP,LM>
  AMD Features2=0x1d4037ff<LAHF,CMP,SVM,ExtAPIC,CR8,ABM,SSE4A,MAS,Prefetch,OSVW,IBS,SKINIT,WDT,Topology,PNXC,DBE,PTSC,PL2I>
  Structured Extended Features=0x8<BMI1>
  XSAVE Features=0x1<XSAVEOPT>
  SVM: NP,NRIP,AFlush,DAssist,NAsids=8
  TSC: P-state invariant, performance statistics
real memory  = 4294967296 (4096 MB)
avail memory = 4070649856 (3882 MB)
Event timer "LAPIC" quality 600
ACPI APIC Table: <COREv4 COREBOOT>
FreeBSD/SMP: Multiprocessor System Detected: 4 CPUs
FreeBSD/SMP: 1 package(s) x 4 core(s)
ioapic1: Changing APIC ID to 5
ioapic0 <Version 2.1> irqs 0-23 on motherboard
ioapic1 <Version 2.1> irqs 24-55 on motherboard
SMP: AP CPU #3 Launched!
SMP: AP CPU #2 Launched!
SMP: AP CPU #1 Launched!
Timecounter "TSC" frequency 998149644 Hz quality 1000
wlan: mac acl policy registered
random: entropy device external interface
000.000016 [4213] netmap_init               netmap: loaded module
module_register_init: MOD_LOAD (vesa, 0xffffffff8114aef0, 0) error 19
kbd0 at kbdmux0
nexus0
vtvga0: <VT VGA driver> on motherboard
cryptosoft0: <software crypto> on motherboard
acpi0: <COREv4 COREBOOT> on motherboard
acpi0: Power Button (fixed)
cpu0: <ACPI CPU> on acpi0
cpu1: <ACPI CPU> on acpi0
cpu2: <ACPI CPU> on acpi0
cpu3: <ACPI CPU> on acpi0
atrtc0: <AT realtime clock> port 0x70-0x71 irq 8 on acpi0
atrtc0: registered as a time-of-day clock, resolution 1.000000s
Event timer "RTC" frequency 32768 Hz quality 0
attimer0: <AT timer> port 0x40-0x43 irq 0 on acpi0
Timecounter "i8254" frequency 1193182 Hz quality 0
Event timer "i8254" frequency 1193182 Hz quality 100
Timecounter "ACPI-safe" frequency 3579545 Hz quality 850
acpi_timer0: <32-bit timer at 3.579545MHz> port 0x818-0x81b on acpi0
hpet0: <High Precision Event Timer> iomem 0xfed00000-0xfed003ff on acpi0
Timecounter "HPET" frequency 14318180 Hz quality 950
acpi_button0: <Power Button> on acpi0
pcib0: <ACPI Host-PCI bridge> port 0xcf8-0xcff on acpi0
pcib0: could not evaluate _ADR - AE_NOT_FOUND
pci0: <ACPI PCI bus> on pcib0
pci0: <base peripheral, IOMMU> at device 0.2 (no driver attached)
pcib1: <ACPI PCI-PCI bridge> irq 24 at device 2.1 on pci0
pcib1: failed to allocate initial I/O port window: 0x1000-0x1fff
pci1: <ACPI PCI bus> on pcib1
igb0: <Intel(R) PRO/1000 Network Connection, Version - 2.5.3-k> mem 0xf7800000-0xf781ffff,0xf7820000-0xf7823fff irq 24 at device 0.0 on pci1
igb0: Using MSIX interrupts with 3 vectors
igb0: Ethernet address: 00:0d:b9:50:10:e0
igb0: Bound queue 0 to cpu 0
igb0: Bound queue 1 to cpu 1
igb0: netmap queues/slots: TX 2/1024, RX 2/1024
pcib2: <ACPI PCI-PCI bridge> irq 25 at device 2.2 on pci0
pci2: <ACPI PCI bus> on pcib2
igb1: <Intel(R) PRO/1000 Network Connection, Version - 2.5.3-k> port 0x2000-0x201f mem 0xf7900000-0xf791ffff,0xf7920000-0xf7923fff irq 28 at device 0.0 on pci2
igb1: Using MSIX interrupts with 3 vectors
igb1: Ethernet address: 00:0d:b9:50:10:e1
igb1: Bound queue 0 to cpu 2
igb1: Bound queue 1 to cpu 3
igb1: netmap queues/slots: TX 2/1024, RX 2/1024
pcib3: <ACPI PCI-PCI bridge> irq 26 at device 2.3 on pci0
pci3: <ACPI PCI bus> on pcib3
igb2: <Intel(R) PRO/1000 Network Connection, Version - 2.5.3-k> port 0x3000-0x301f mem 0xf7a00000-0xf7a1ffff,0xf7a20000-0xf7a23fff irq 32 at device 0.0 on pci3
igb2: Using MSIX interrupts with 3 vectors
igb2: Ethernet address: 00:0d:b9:50:10:e2
igb2: Bound queue 0 to cpu 0
igb2: Bound queue 1 to cpu 1
igb2: netmap queues/slots: TX 2/1024, RX 2/1024
pcib4: <ACPI PCI-PCI bridge> irq 27 at device 2.4 on pci0
pci4: <ACPI PCI bus> on pcib4
igb3: <Intel(R) PRO/1000 Network Connection, Version - 2.5.3-k> port 0x4000-0x401f mem 0xf7b00000-0xf7b1ffff,0xf7b20000-0xf7b23fff irq 36 at device 0.0 on pci4
igb3: Using MSIX interrupts with 3 vectors
igb3: Ethernet address: 00:0d:b9:50:10:e3
igb3: Bound queue 0 to cpu 2
igb3: Bound queue 1 to cpu 3
igb3: netmap queues/slots: TX 2/1024, RX 2/1024
pcib5: <ACPI PCI-PCI bridge> irq 24 at device 2.5 on pci0
pci5: <ACPI PCI bus> on pcib5
ath0: <Atheros 9280> mem 0xf7c00000-0xf7c0ffff irq 40 at device 0.0 on pci5
[ath] enabling AN_TOP2_FIXUP
ath0: [HT] enabling HT modes
ath0: [HT] 1 stream STBC receive enabled
ath0: [HT] 1 stream STBC transmit enabled
ath0: [HT] 2 RX streams; 2 TX streams
ath0: AR9280 mac 128.2 RF5133 phy 13.0
ath0: 2GHz radio: 0x0000; 5GHz radio: 0x00c0
pci0: <encrypt/decrypt> at device 8.0 (no driver attached)
xhci0: <AMD FCH USB 3.0 controller> mem 0xf7f22000-0xf7f23fff irq 18 at device 16.0 on pci0
xhci0: 32 bytes context size, 64-bit DMA
xhci0: Unable to map MSI-X table 
usbus0 on xhci0
usbus0: 5.0Gbps Super Speed USB v3.0
ahci0: <AMD Hudson-2 AHCI SATA controller> port 0x5010-0x5017,0x5020-0x5023,0x5018-0x501f,0x5024-0x5027,0x5000-0x500f mem 0xf7f25000-0xf7f253ff at device 17.0 on pci0
ahci0: AHCI v1.30 with 2 6Gbps ports, Port Multiplier supported with FBS
ahcich0: <AHCI channel> at channel 0 on ahci0
ahcich1: <AHCI channel> at channel 1 on ahci0
ehci0: <AMD FCH USB 2.0 controller> mem 0xf7f26000-0xf7f260ff irq 18 at device 19.0 on pci0
usbus1: EHCI version 1.0
usbus1 on ehci0
usbus1: 480Mbps High Speed USB v2.0
isab0: <PCI-ISA bridge> at device 20.3 on pci0
isa0: <ISA bus> on isab0
sdhci_pci0: <Generic SD HCI> mem 0xf7f27000-0xf7f270ff at device 20.7 on pci0
sdhci_pci0: 1 slot(s) allocated
uart0: <16550 or compatible> port 0x3f8-0x3ff irq 4 flags 0x10 on acpi0
uart0: console (115200,n,8,1)
uart1: <16550 or compatible> port 0x2f8-0x2ff irq 3 on acpi0
orm0: <ISA Option ROM> at iomem 0xee800-0xeffff on isa0
ppc0: cannot reserve I/O port range
hwpstate0: <Cool`n'Quiet 2.0> on cpu0
ZFS NOTICE: Prefetch is disabled by default if less than 4GB of RAM is present;
            to enable, add "vfs.zfs.prefetch_disable=0" to /boot/loader.conf.
ZFS filesystem version: 5
ZFS storage pool version: features support (5000)
Timecounters tick every 1.000 msec
ugen1.1: <AMD EHCI root HUB> at usbus1
ugen0.1: <0x1022 XHCI root HUB> at usbus0
uhub0: <AMD EHCI root HUB, class 9/0, rev 2.00/1.00, addr 1> on usbus1
uhub1: <0x1022 XHCI root HUB, class 9/0, rev 3.00/1.00, addr 1> on usbus0
ada0 at ahcich0 bus 0 scbus0 target 0 lun 0
ada0: <ST4000LM024-2AN17V 0001> ACS-3 ATA SATA 3.x device
ada0: Serial Number WCJ06GJ8
ada0: 600.000MB/s transfers (SATA 3.x, UDMA6, PIO 8192bytes)
ada0: Command Queueing enabled
ada0: 3815447MB (7814037168 512 byte sectors)
ada1 at ahcich1 bus 0 scbus1 target 0 lun 0
ada1: <ST4000LM024-2AN17V 0001> ACS-3 ATA SATA 3.x device
ada1: Serial Number WCK0NTGF
ada1: 600.000MB/s transfers (SATA 3.x, UDMA6, PIO 8192bytes)
ada1: Command Queueing enabled
ada1: 3815447MB (7814037168 512 byte sectors)
Trying to mount root from ufs:/dev/gpt/rootfs [rw,noatime]...
uhub1: 4 ports with 4 removable, self powered
Root mount waiting for: usbus1 usbus0
uhub0: 2 ports with 2 removable, self powered
ugen0.2: <SEM USB Keyboard> at usbus0
ukbd0 on uhub1
ukbd0: <SEM USB Keyboard, class 0/0, rev 1.10/1.10, addr 1> on usbus0
kbd1 at ukbd0
ugen1.2: <vendor 0x0438 product 0x7900> at usbus1
uhub2 on uhub0
uhub2: <vendor 0x0438 product 0x7900, class 9/0, rev 2.00/0.18, addr 2> on usbus1
uhub2: 4 ports with 4 removable, self powered
Root mount waiting for: usbus1
Root mount waiting for: usbus1
Root mount waiting for: usbus1
ugen1.3: <Atrue UDOM> at usbus1
umass0 on uhub2
umass0: <Atrue UDOM, class 0/0, rev 2.00/11.00, addr 3> on usbus1
umass0:  SCSI over Bulk-Only; quirks = 0x8100
umass0:2:0: Attached to scbus2
da0 at umass-sim0 bus 0 scbus2 target 0 lun 0
da0: <Atrue UDOM 1100> Removable Direct Access SPC-2 SCSI device
da0: Serial Number 201302IP00029LEQBOOC
da0: 40.000MB/s transfers
da0: 15480MB (31703040 512 byte sectors)
da0: quirks=0x2<NO_6_BYTE>
mountroot: waiting for device /dev/gpt/rootfs...
WARNING: /mnt was not properly dismounted
WARNING: /mnt: mount pending error: blocks 8 files 3
Mounting filesystems...
tunefs: soft updates remains unchanged as enabled
WARNING: /mnt: reload pending error: blocks 8 files 3
tunefs: file system reloaded
camcontrol: ATA ATAPI_IDENTIFY via pass_16 failed
** /dev/gpt/rootfs
** Last Mounted on /mnt
** Root file system
** Phase 1 - Check Blocks and Sizes
INCORRECT BLOCK COUNT I=1204396 (80 should be 72)
CORRECT? yes

** Phase 2 - Check Pathnames
** Phase 3 - Check Connectivity
** Phase 4 - Check Reference Counts
LINK COUNT DIR I=2  OWNER=root MODE=40755
SIZE=512 MTIME=Mar  1 04:38 2020  COUNT 25 SHOULD BE 23
ADJUST? yes

LINK COUNT DIR I=1203862  OWNER=root MODE=40755
SIZE=1536 MTIME=Mar  1 04:39 2020  COUNT 13 SHOULD BE 12
ADJUST? yes

UNREF FILE  I=1605141  OWNER=root MODE=100644
SIZE=0 MTIME=Mar  1 04:38 2020 
RECONNECT? yes

** Phase 5 - Check Cyl groups
FREE BLK COUNT(S) WRONG IN SUPERBLK
SALVAGE? yes

SUMMARY INFORMATION BAD
SALVAGE? yes

BLK(S) MISSING IN BIT MAPS
SALVAGE? yes

65311 files, 444171 used, 3135771 free (1675 frags, 391762 blocks, 0.0% fragmentation)

***** FILE SYSTEM STILL DIRTY *****

***** FILE SYSTEM WAS MODIFIED *WARNING: /mnt: reload pending error: blocks 8 files 3
****

***** PLEASE RERUN FSCK *****
** /dev/gpt/rootfs
** Last Mounted on /mnt
** Root file system
** Phase 1 - Check Blocks and Sizes
** Phase 2 - Check Pathnames
** Phase 3 - Check Connectivity
** Phase 4 - Check Reference Counts
** Phase 5 - Check Cyl groups
65311 files, 444171 used, 3135771 free (1675 frags, 391762 blocks, 0.0% fragmentation)

***** FILE SYSTEM MARKED CLEAN *****
random: unblocking device.
Setting hostuuid: ab14bf97-5172-11ea-9e9f-000db95010e0.
Setting hostid: 0x68236e2a.
Configuring vt: blanktime.
Configuring crash dump device: /dev/null
.ELF ldconfig path: /lib /usr/lib /usr/local/lib /usr/local/lib/ipsec /usr/local/lib/perl5/5.30/mach/CORE /usr/local/lib/samba4
32-bit compatibility ldconfig path:
done.
>>> Invoking early script 'update'
WARNING: /usr/local/etc/pkg/repos/OPNsense.conf does not use fingerprints, disabling signature checks.
WARNING: /usr/local/etc/pkg/repos/OPNsense.conf does not use fingerprints, disabling signature checks.
>>> Invoking early script 'configd'
Starting configd.
>>> Invoking early script 'templates'
Generating configuration: OK
>>> Invoking early script 'backup'
>>> Invoking backup script 'captiveportal'
>>> Invoking backup script 'dhcpleases'
>>> Invoking backup script 'duid'
>>> Invoking backup script 'netflow'
>>> Invoking backup script 'rrd'
>>> Invoking early script 'carp_maintenance'
>>> Invoking early script 'carp_service_status'
OK
Launching the init system...done.
Initializing...........done.
Starting device manager...done.
Configuring login behaviour...done.
Configuring loopback interface...done.
Configuring kernel modules...done.
Setting up extended sysctls...sysctl: oid 'kern.randompid' is read only
sysctl: oid 'kern.randompid' is read only
done.
Setting timezone...done.
Writing firmware setting...done.
Writing trust files...done.
Setting hostname: host.example.com
Generating /etc/hosts...done.
Configuring system logging...done.
Configuring loopback interface...done.
Creating wireless clone interfaces...done.
Configuring LAN1 interface...done.
Configuring LAN2 interface...done.
Configuring LAN3 interface...done.
Configuring bridge interfaces (1)...done.
Configuring bridge interfaces (2)...done.
Configuring BRIDGE interface...done.
Configuring WIFIWAN interface...done.
Creating IPsec VTI instances...done.
Creating OpenVPN instances...done.
Generating /etc/resolv.conf...done.
Configuring firewall.......done.
Starting PFLOG...done.
Configuring OpenSSH...done.
Starting web GUI...done.
Configuring CRON...done.
Setting up routes...done.
Generating /etc/hosts...done.
Starting DHCPv4 service...done.
Starting Unbound DNS...done.
Setting up gateway monitors...done.
Configuring firewall.......done.
Starting PFLOG...done.
Syncing OpenVPN settings...done.
Configuring dynamic DNS clients...done.
Starting NTP service...deferred.
Starting Unbound DNS...done.
Generating RRD graphs...done.
Configuring system logging...done.
>>> Invoking start script 'newwanip'
Reconfiguring IPv4 on ath0_wlan1: OK
Reconfiguring IPv4 on ovpns1: OK
Reconfiguring IPv4 on ovpnc2: OK
>>> Invoking start script 'freebsd'
Performing sanity check on Samba configuration: FAILED
Starting monit.
Starting Monit 5.26.0 daemon with http interface at /var/run/monit.sock
>>> Invoking start script 'wireguard'
Setting up routes...done.
Setting up gateway monitors...done.
Configuring firewall.......done.
Starting PFLOG...done.
>>> Invoking start script 'carp'
>>> Invoking start script 'cron'
Starting Cron: OK
>>> Invoking start script 'beep'
Root file system: /dev/gpt/rootfs
Sun Mar  1 18:15:40 UTC 2020

*** host.example.com: OPNsense 20.1.1 (amd64/OpenSSL) ***

 BRIDGE (bridge0) -> v4: 10.100.40.1/24
 LAN1 (igb1)     -> 
 LAN2 (igb2)     -> 
 LAN3 (igb3)     -> 
 WIFIWAN (ath0_wlan1) -> v4/DHCP4: 192.168.1.101/24

 HTTPS: SHA256 XXXXXXXXXXXXXX
 SSH:   SHA256 XXXXXXXXXXXXXX (ECDSA)
 SSH:   SHA256 XXXXXXXXXXXXXX (ED25519)
 SSH:   SHA256 XXXXXXXXXXXXXX (RSA)

FreeBSD/amd64 (host.example.com) (ttyu0)

Last login: Sun Mar  1 03:56:37 on ttyu0
----------------------------------------------
|      Hello, this is OPNsense 20.1          |         @@@@@@@@@@@@@@@
|                                            |        @@@@         @@@@
| Website:      https://opnsense.org/        |         @@@\\\   ///@@@
| Handbook:     https://docs.opnsense.org/   |       ))))))))   ((((((((
| Forums:       https://forum.opnsense.org/  |         @@@///   \\\@@@
| Lists:        https://lists.opnsense.org/  |        @@@@         @@@@
| Code:         https://github.com/opnsense  |         @@@@@@@@@@@@@@@
----------------------------------------------

*** arch.circlenaut.com: OPNsense 20.1.1 (amd64/OpenSSL) ***

 BRIDGE (bridge0) -> v4: 10.100.40.1/24
 LAN1 (igb1)     -> 
 LAN2 (igb2)     -> 
 LAN3 (igb3)     -> 
 WIFIWAN (ath0_wlan1) -> v4/DHCP4: 192.168.1.101/24

 HTTPS: SHA256 XXXXXXXXXXXXXXX
 SSH:   SHA256 XXXXXXXXXXXXXXX (ECDSA)
 SSH:   SHA256 XXXXXXXXXXXXXXX (ED25519)
 SSH:   SHA256 XXXXXXXXXXXXXXX (RSA)

  0) Logout                              7) Ping host
  1) Assign interfaces                   8) Shell
  2) Set interface IP address            9) pfTop
  3) Reset the root password            10) Firewall log
  4) Reset to factory defaults          11) Reload all services
  5) Power off system                   12) Update from console
  6) Reboot system                      13) Restore a backup

Enter an option: 5

The system will halt and power off. Do you want to proceed? [y/N]: y

>>> Invoking stop script 'beep'
>>> Invoking stop script 'freebsd'
smbd not running? (check /var/run/samba4/smbd.pid).
nmbd not running? (check /var/run/samba4/nmbd.pid).
Stopping monit.
Waiting for PIDS: 62345.
>>> Invoking stop script 'backup'
>>> Invoking backup script 'captiveportal'
>>> Invoking backup script 'dhcpleases'
>>> Invoking backup script 'duid'
>>> Invoking backup script 'netflow'
>>> Invoking backup script 'rrd'
>>> Invoking stop script 'config'
shutdown: [pid 48511]
Shutdown NOW!
                                                                               
*** FINAL System shutdown message from root@arch.circlenaut.com ***          

System going down IMMEDIATELY                                                  

                                                                               
Mar  1 18:18:19 lighttpd[6421]: (server.c.1970) server stopped by UID = 0 PID = 48511 
ovpnc2: link state changed to DOWN
Waiting (max 60 seconds) for system process `vnlru' to stop... done
Waiting (max 60 seconds) for system process `bufdaemon' to stop... done
Waiting (max 60 seconds) for system process `syncer' to stop... 
Syncing disks, vnodes remaining... 4 3 1 0 0 0 done
All buffers synced.
Uptime: 7m21s
ukbd0: detached
uhid0: detached
umass0: detached
uhub2: detached
acpi0: Powering system off

Terminating...
Skipping tty reset...
Thanks for using picocom
```
